pkg/services/ngalert/remote/alertmanager.go

```diff
@@ -247,10 +247,45 @@ func (am *Alertmanager) TestTemplate(ctx context.Context, c apimodels.TestTempla
 }
 
 func (am *Alertmanager) StopAndWait() {
+	am.sender.Stop()
 }
 
+// TODO: change implementation, this is only useful for testing other methods
 func (am *Alertmanager) Ready() bool {
-	return false
+	readyURL := strings.TrimSuffix(am.url, "/") + "/-/ready"
```

Comment:
I think we should move all of this ready code to `ApplyConfig`  and use a bool to know wether we're ready or not. 

`Ready` is called everytime we try to get an alertmanager (even on the API) so I don't think we should be making HTTP calls wether this is called.